kliff/dataset/dataset.py

```diff
@@ -301,67 +499,207 @@ def order_by_species(self):
+            raise DatasetError(f"{colabfit_dataset} is either empty or does not exist")
+
+        configs = []
+        for do in dataset_dos:
+            co_doc = database_client.configurations.find_one(
+                {"relationships.data_objects": {"$in": [do]}}
+            )
+            pi_doc = database_client.property_instances.find(
+                {"relationships.data_objects": {"$in": [do]}}
+            )
+            co_id = co_doc["colabfit-id"]
+            pi_ids = [i["colabfit-id"] for i in pi_doc]
 
-        configs = self._read(path, weight, self.file_format)
-        self.configs.extend(configs)
+            configs.append(
+                Configuration.from_colabfit(database_client, co_id, pi_ids, weight)
+            )
+            # TODO: reduce number of queries to database. Current: 4 per configuration
 
-    def get_configs(self) -> List[Configuration]:
+        if len(configs) <= 0:
+            raise DatasetError(f"No dataset file with in {colabfit_dataset} dataset.")
+
+        logger.info(f"{len(configs)} configurations read from {colabfit_dataset}")
+
+        return configs
+
+    def add_from_colabfit(
+        self,
+        colabfit_database: str = None,
+        colabfit_dataset: str = None,
+        weight: Optional[Weight] = None,
+    ):
         """
-        Get the configurations.
+        Read configurations from colabfit database and add them to the dataset.
+
+        Args:
+            colabfit_database: Name of the colabfit Mongo database to read from.
+            colabfit_dataset: Name of the colabfit dataset instance to read from.
+            weight: an instance that computes the weight of the configuration in the loss
+                function.
+
         """
-        return self.configs
+        if colabfit_database is not None:
+            if colabfit_dataset is not None:
+                # open link to the mongo
+                if colabfit_installed:
+                    mongo_client = MongoDatabase(colabfit_database)
+                    colabfit_dataset = colabfit_dataset
+                    configs = Dataset._read_from_colabfit(
+                        mongo_client, colabfit_dataset, weight
+                    )
+                    self.configs.extend(configs)
+                else:
+                    logger.error(f"colabfit tools not installed.")
+                    raise DatasetError(
+                        f"You are trying to read configuration from colabfit dataset"
+                        " but colabfit-tools module is not installed."
+                        " Please do `pip install colabfit` first"
+                    )
+            else:
+                logger.error(
+                    f"colabfit database provided ({colabfit_database}) but not dataset."
+                )
+                raise DatasetError(f"No dataset ID given.")
 
-    def get_num_configs(self) -> int:
+    @classmethod
+    def from_path(
+        cls,
+        path: Optional[Path] = None,
```

Comment:
`path` should not be `None`, either `Optional`

Reply:
`path` is now a positional argument. Converted from `Optional[Path]` to `Union[Path, str]`